salt/modules/upstart.py

```diff
@@ -148,13 +151,14 @@ def _runlevel():
     '''
     if 'upstart._runlevel' in __context__:
         return __context__['upstart._runlevel']
-    out = __salt__['cmd.run'](['runlevel', '{0}'.format(_find_utmp())], python_shell=False)
-    try:
-        ret = out.split()[1]
-    except IndexError:
-        # The runlevel is unknown, return the default
-        ret = _default_runlevel()
-    __context__['upstart._runlevel'] = ret
+    ret = _default_runlevel()
+    utmp = _find_utmp()
+    if utmp:
+        out = __salt__['cmd.run'](['runlevel', '{0}'.format(utmp)], python_shell=False)
+        try:
+            ret = out.split()[1]
+        except IndexError:
+            __context__['upstart._runlevel'] = ret
```

Comment:
Can you elaborate on why the `__context__` setting here has been moved into the exception? Everything up until that point looks good, but I am unsure why only the IndexError updates the context.

Reply:
Thanks for spotting this. This was fixed (restored original behavior) + updated.